src/context.ts

```diff
@@ -0,0 +1,50 @@
+import React, { createContext, useContext, useRef } from 'react'
+import create, { UseStore } from './index'
+import { EqualityChecker, State, StateCreator, StateSelector } from './vanilla'
+
+function createZustand<TState extends State>(initialState?: TState) {
+  const ZustandContext = createContext<UseStore<TState> | undefined>(undefined)
+
+  const Provider = ({
+    createState,
+    children,
+  }: {
+    createState: StateCreator<TState>
+    children: React.ReactNode
+  }) => {
+    const storeRef = useRef() as React.MutableRefObject<UseStore<TState>>
```

Comment:
```suggestion
    const storeRef = useRef<UseStore<TState>>()
```
I prefer to annotate type instead of assertion here and assert type in L23, but it's not a big deal.

Reply:
@dai-shi Ah.. createState prop on Provider alone is not going to work for another reason.

Have a look at my last [next.js solution](https://codesandbox.io/s/nextjs-with-zustand-forked-sr6z4?file=/lib/store.js:1266-1450) useHydrate() hook. (It is a working example, with all the bugs fixed I mentioned at #182).

See the logic to merge new states into existing store on client-side? It is unavoidable, and has to be there for next.js to work properly. Our current Provider interface is not doing this, nor does it expose the store.

So I am in favor of passing store. Property name can be either `store` or maybe `initialStore`?